webrtc/README.md

```diff
@@ -0,0 +1,268 @@
+- Replace TURN with libp2p's Circuit Relay v2
+  - https://github.com/libp2p/specs/blob/master/relay/circuit-v2.md
+- Use DCUtR over Circuit Relay v2 to transmit SDP information
+  1. Transform ICE candidates in SDP to multiaddresses.
+  2. Transmit the set of multiaddresses to the remote via DCUtR.
+  3. Transform the set of multiaddresses back to the remotes SDP.
+  4. https://github.com/libp2p/specs/blob/master/relay/DCUtR.md
+
+#### Open Questions
+
+- Can _Browser_ know upfront its UDP port which it is listening for incoming
+  connections on? Does the browser reuse the UDP port across many WebRTC
+  connections? If that is the case one could connect to any public node, with
+  the remote telling the local node what port it is perceived on.
+
+- Can _Browser_ control the lifecycle of its local TLS certificate, i.e. can
+  _Browser_ use the same TLS certificate for multiple WebRTC connections?
+
+  Yes. For the lifetime of the page, one can generate a certificate once and
+  reuse it across connections. See also
+  https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/RTCPeerConnection#using_certificates
+
+- Can two _Browsers_ exchange their SDP packets via a third server node using
+  Circuit Relay v2 and DCUtR? Instead of exchanging the original SDP packets,
+  could they exchange their multiaddr and construct the remote's SDP packet
+  based on it?
+
+## Connection Security
+
+While WebRTC offers confidentiality and integrity via TLS, one still needs to
+authenticate the remote peer by its libp2p identity.
+
+After [Connection Establishment](#connection-establishment):
+
+1. _A_ opens a WebRTC datachannel and starts a Noise handshake using _A_'s and
+   _B_'s libp2p identity. See
+   [noise-libp2p](https://github.com/libp2p/specs/tree/master/noise).
+
+2. _A_ and _B_ write their TLS certificate fingerprint on the negotiated Noise channel.
+
+3. _A_ and _B_ read the other sides TLS certificate fingerprint on the
+   negotiated Noise channel and compare it to the ones verified during the DTLS
+   handshake.
+
+### Open Questions
+
+- Can a _Browser_ access the fingerprint of its TLS certificate?
+
+  Chrome allows you to access the fingerprint of any locally-created certificate
+  directly via `RTCCertificate#getFingerprints`. Firefox does not allow you to
+  do so. Browser compatibility can be found
+  [here](https://developer.mozilla.org/en-US/docs/Web/API/RTCCertificate). In
+  practice, this is not an issue since the fingerprint is embedded in the local
+  SDP string.
+
+- Is the above proposed additional handshake secure? See also alternative
+  proposed Handshake for
+  [WebTransport](https://github.com/libp2p/specs/pull/404).
+
+- On the server side, can one derive the TLS certificate in a deterministic way
+  based on a node's libp2p private key? Benefit would be that a node only needs
+  to persist the libp2p private key and not the TLS key material while still
+  maintaining a fixed TLS certificate fingerprint.
```

Comment:
Yes, this is trivially possible. We don't need to specify anything though, since this is purely an implementation decision.

Reply:
Removed with https://github.com/libp2p/specs/pull/412/commits/16e38fbb1646e9c97d0a169a946a6e7126cd7421.